### PR TITLE
(CDAP-7719) Use TMS for audit log publishing

### PR DESCRIPTION
**Pull request by chtyim**
- Also removed Kafka and ZK from SDK.

### CODE REVIEW

**Comment by chtyim** [comment]
Building in http://builds.cask.co/browse/CDAP-RUT395

File: cdap-common/src/main/resources/cdap-default.xml [hunk 1]
```diff
@@ -405,18 +405,25 @@
...
     </description>
   </property>
 
   <property>
-    <name>audit.kafka.topic</name>
+    <name>audit.topic</name>
     <value>audit</value>
     <description>
-      Apache Kafka topic name to which audit messages are published
+      The topic name in the messaging system to which audit messages are published
```
Comment (gokulavasan) [inline cdap-common/src/main/resources/cdap-default.xml:416]: Should we mention that the namespace for this topic will always be "system" ?

File: cdap-master/src/main/java/co/cask/cdap/master/startup/ConfigurationCheck.java [hunk 2]
```diff
@@ -134,8 +136,11 @@ private void checkKafkaTopic(Set<String> problemKeys) {
     if (!isValidKafkaTopic(Constants.Logging.KAFKA_TOPIC)) {
       problemKeys.add(Constants.Logging.KAFKA_TOPIC);
     }
-    if (!isValidKafkaTopic(Constants.Audit.KAFKA_TOPIC)) {
-      problemKeys.add(Constants.Audit.KAFKA_TOPIC);
+  }
+
+  private void checkMessagingTopics(Set<String> problemKeys) {
```
Comment (gokulavasan) [inline cdap-master/src/main/java/co/cask/cdap/master/startup/ConfigurationCheck.java]: Do we need to check if all of the system topics (from the comma separated list from cConf) to be created are all valid IDs?

File: cdap-tms/src/main/java/co/cask/cdap/messaging/MessagingServices.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+          throw e;
+        }
+        if (delay > 0) {
+          TimeUnit.MILLISECONDS.sleep(delay);
+        }
+      }
+    }
+  }
+
+
```
Comment (gokulavasan) [inline cdap-tms/src/main/java/co/cask/cdap/messaging/MessagingServices.java]: nit: remove extra new line

File: cdap-tms/src/main/java/co/cask/cdap/messaging/service/CoreMessagingService.java [hunk 4]
```diff
@@ -204,6 +218,32 @@ public void rollback(TopicId topicId, RollbackDetail rollbackDetail) throws Topi
...
   protected void startUp() throws Exception {
+    Queue<TopicId> asyncCreationTopics = new LinkedList<>();
+
+    for (String topic : new HashSet<>(cConf.getTrimmedStringCollection(Constants.MessagingSystem.SYSTEM_TOPICS))) {
+      TopicId topicId;
+      try {
+        topicId = NamespaceId.SYSTEM.topic(topic);
+      } catch (IllegalArgumentException e) {
+        // Ignore invalid topic
+        LOG.warn("Ignore creation of invalid topic '{}'.", topic);
```
Comment (gokulavasan) [inline cdap-tms/src/main/java/co/cask/cdap/messaging/service/CoreMessagingService.java]: Ah ok , I see we are making the check here. Ignore my previous comment about configuration check of SYSTEM_TOPICS property.

File: cdap-docs/tracker-manual/source/index.rst [hunk 5]
```diff
@@ -165,19 +165,16 @@ in the ``cdap-site.xml`` as described above, need to follow these steps:
...
-  Audit Log reader configuration (the property ``auditLogKafkaConfig``). It is the Kafka
-  Consumer Flowlet configuration information. For example::
+- Create an application configuration file (``appconfig.txt``) that contains the
+  Audit Log reader configuration (the property ``auditLogConfig``). For example::
     
     {
       "config": {
-        "auditLogKafkaConfig": {
-          "zookeeperString": "<host>:<port>/cdap/kafka"
+        "auditLogConfig" : {
```
Comment (gokulavasan) [inline cdap-docs/tracker-manual/source/index.rst:173]: Should we leave zookeeperString property in there (or mention that it is required in distributed mode?) ?

**Comment by gokulavasan** [comment]
Done with review. Minor comments. Rest LGTM 👍

File: cdap-common/src/main/resources/cdap-default.xml (hunk 1 above)
Comment (chtyim) [reply cdap-common/src/main/resources/cdap-default.xml:416]: I think everything we have here are for the CDAP system settings, that's why I don't say it (as not to confuse user as well).

**Comment by chtyim** [comment]
Build passed. Squashed commits, rebased on release/4.0 to resolve conflict on the md5 hash of cdap-default.xml

**Pull request merged**